byron/ledger/executable-spec/src/Ledger/UTxO.hs

```diff
@@ -31,27 +31,20 @@ import Ledger.Update (PParams (PParams), _factorA, _factorB)
 
 -- |A unique ID of a transaction, which is computable from the transaction.
 newtype TxId = TxId { getTxId :: Hash }
-  deriving stock (Show)
+  deriving stock (Show, Generic)
   deriving newtype (Eq, Ord, Hashable)
-
-instance HasTypeReps TxId where
-  typeReps x = typeOf x <| typeOf (getTxId x) <| empty
+  deriving anyclass (HasTypeReps)
 
 -- |The input of a UTxO.
 --
 --     * __TODO__ - is it okay to use list indices instead of implementing the Ix Type?
-data TxIn = TxIn TxId Natural deriving (Show, Eq, Ord, Generic, Hashable)
-
-instance HasTypeReps TxIn where
-  typeReps x@(TxIn i' n) = typeOf x <| typeOf i' <| typeOf n <| empty
+data TxIn = TxIn TxId Natural
```

Comment:
Don't we need `derive anyclass` here as well?

Reply:
this is ok because it's _data TxIn_ and not a new type

Reply:
Oh, silly me :man_facepalming: